static/app/components/avatarChooser.tsx

```diff
@@ -114,22 +114,22 @@ class AvatarChooser extends React.Component<Props, State> {
   handleSaveSettings = (ev: React.MouseEvent) => {
     const {endpoint, api, type} = this.props;
     const {model, dataUrl} = this.state;
-    const isSentryApp = type?.startsWith('sentryApp');
 
     ev.preventDefault();
-    const avatarType = model && model.avatar ? model.avatar.avatarType : undefined;
-    const avatarPhoto = dataUrl ? dataUrl.split(',')[1] : undefined;
+    const avatarType = model?.avatar?.avatarType;
+    const avatarPhoto = dataUrl?.split(',')[1];
 
     const data: {
```

Comment:
I'm not sure if this would be valid typescript but i think a conditional would be clearer than the ternary.
```tsx
const data: {
  avatar_photo?: string;
  avatar_type?: string;
  color?: boolean;
} = {avatar_type: avatarType};

// If an image has been uploaded, then another option is selected, we should not submit the image.
if (avatarType === 'upload') {
  data.avatar_photo = avatarPhoto;
}
```

Reply:
good catch, implemented